### PR TITLE
Save env state on deploy

### PR DESCRIPTION
**Pull request by ragurney**

### CODE REVIEW

File: app/models/deploy.rb [hunk 1]
```diff
@@ -266,4 +268,18 @@ def deploy_buddy
   def trim_reference
     self.reference = reference.strip if reference.present?
   end
+
+  def serialize_env_state
```
Comment (grosser) [inline app/models/deploy.rb]: stage has a cheaper/compact env serialization :(

Comment (grosser) [reply app/models/deploy.rb]: https://github.com/zendesk/samson/blob/master/plugins/env/app/decorators/project_decorator.rb#L22

... also it should not blow up when `env` plugin is not enabled :trollface:

Comment (grosser) [reply app/models/deploy.rb]: having the hash would make deploying an old deploy easier ... unless there is a new deploy-group 🤷‍♂️

Comment (ragurney) [reply app/models/deploy.rb]: Where?

**Comment by grosser** [comment]
test

**Comment by grosser** [comment]
- make matches_scope public if it's used from the outside instead of `send`

 - serialized_environment_variables can be private ?

 - variables_to_string -> serialize ?

**Pull request merged**